File: python/labs/lab1/student.py
class Student:
    def __init__(self):
        self.student_id = 0
        self.student_name = ""
        self.city = ""
        self.marks1 = 0
        self.marks2 = 0
        self.marks3 = 0
        self.fee_per_month = 0.0
        self.is_eligible_for_scholarship = False

    # Methods
    def get_annual_fee(self):
        return self.fee_per_month * 12

    def get_total_marks(self):
        return self.marks1 + self.marks2 + self.marks3

    def get_average(self):
        return self.get_total_marks() / 3

    def get_result(self):
        return "pass" if self.marks1 > 60 and self.marks2 > 60 and self.marks3 > 60 else "fail"


# --- Test Script ---
s1 = Student()
s1.student_name = "Alice"
s1.marks1, s1.marks2, s1.marks3 = 85, 90, 78
s1.fee_per_month = 5000
s1.is_eligible_for_scholarship = True

s2 = Student()
s2.student_name = "Bob"
s2.marks1, s2.marks2, s2.marks3 = 70, 65, 72
s2.fee_per_month = 4500
s2.is_eligible_for_scholarship = False

s3 = Student()
s3.student_name = "Charlie"
s3.marks1, s3.marks2, s3.marks3 = 95, 92, 88
s3.fee_per_month = 5500
s3.is_eligible_for_scholarship = True

students = [s1, s2, s3]

# Highest total marks
highest = max(students, key=lambda s: s.get_total_marks())
print("Highest Marks:", highest.student_name)

# Lowest fee
lowest = min(students, key=lambda s: s.fee_per_month)
print("Lowest Fee:", lowest.student_name, "-", lowest.fee_per_month)

# Each student's details
print("\n--- Student Details ---")
for s in students:
    print(f"Name: {s.student_name}")
    print(f"Total Marks: {s.get_total_marks()}")
    print(f"Average: {s.get_average():.2f}")
    print(f"Result: {s.get_result()}")
    print(f"Scholarship: {s.is_eligible_for_scholarship}")
    print("-" * 30)

File: assign/src/com/task1/Student.java
package com.task1;

public class Student {
    private int studentId;
    private String studentName;
    private int marks1, marks2, marks3;
    private float feePerMonth;
    private boolean isEligibleForScholarship;

    public Student(int id, String name, int m1, int m2, int m3, float fee, boolean scholarship) {
        this.studentId = id;
        this.studentName = name;
        this.marks1 = m1;
        this.marks2 = m2;
        this.marks3 = m3;
        this.feePerMonth = fee;
        this.isEligibleForScholarship = scholarship;
    }

    public int getTotalMarks() { return marks1 + marks2 + marks3; }
    public float getAverage() { return getTotalMarks() / 3.0f; }
    public String getResult() { return (marks1 > 60 && marks2 > 60 && marks3 > 60) ? "pass" : "fail"; }
    public String getStudentName() { return studentName; }
    public float getFeePerMonth() { return feePerMonth; }
    public boolean isEligibleForScholarship() { return isEligibleForScholarship; }
}

// Make com.task1.TestMain non-public
class TestMain {
    public static void main(String[] args) {
        Student[] students = {
                new Student(1, "Alice", 70, 85, 90, 5000, true),
                new Student(2, "Bob", 65, 70, 75, 3000, false),
                new Student(3, "Charlie", 90, 95, 85, 4000, true)
        };

        Student top = students[0];
        Student minFee = students[0];
        for (Student s : students) {
            if (s.getTotalMarks() > top.getTotalMarks()) top = s;
            if (s.getFeePerMonth() < minFee.getFeePerMonth()) minFee = s;
        }

        System.out.println("Highest marks: " + top.getStudentName());
        System.out.println("Least fee: " + minFee.getStudentName() + ", Fee: " + minFee.getFeePerMonth());

        System.out.println("\n--- com.task1.Student Details ---");
        for (Student s : students) {
            System.out.println("Name: " + s.getStudentName() +
                    ", Total: " + s.getTotalMarks() +
                    ", Avg: " + s.getAverage() +
                    ", Result: " + s.getResult() +
                    ", Scholarship: " + s.isEligibleForScholarship());
        }
    }
}